// ng-yandex-metrika/esm5/lib/ng-yandex-metrika-config-factories.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { isPlatformBrowser } from '@angular/common';
import { YandexCounterConfig } from './ng-yandex-metrika.config';
/**
 * @param {?} counterConfigs
 * @param {?=} defaultCounter
 * @return {?}
 */
export function defaultCounterIdFactory(counterConfigs, defaultCounter) {
    return defineDefaultId(counterConfigs, defaultCounter);
}
/**
 * @param {?} counterConfigs
 * @param {?=} defaultCounter
 * @return {?}
 */
export function defineDefaultId(counterConfigs, defaultCounter) {
    /** @type {?} */
    var configs;
    if (counterConfigs instanceof Array) {
        configs = counterConfigs;
    }
    else {
        configs = [(/** @type {?} */ (counterConfigs))];
    }
    /** @type {?} */
    var defaultId;
    if (!defaultCounter) {
        defaultId = configs[0].id;
    }
    else if (typeof defaultCounter === 'number' && defaultCounter < configs.length) {
        defaultId = configs[defaultCounter].id;
    }
    else {
        defaultId = defaultCounter;
    }
    if (!defaultId) {
        console.warn('You provided wrong counter id as a default:', defaultCounter);
        return;
    }
    /** @type {?} */
    var defaultCounterExists = false;
    /** @type {?} */
    var config;
    for (var i = 0; i < configs.length; i++) {
        config = configs[i];
        if (!config.id) {
            console.warn('You should provide counter id to use Yandex metrika counter', config);
            continue;
        }
        if (config.id === defaultId) {
            defaultCounterExists = true;
        }
    }
    if (!defaultCounterExists) {
        console.warn('You provided wrong counter id as a default:', defaultCounter);
    }
    return defaultId;
}
/**
 * @param {?} configs
 * @return {?}
 */
export function countersFactory(configs) {
    return createConfigs(configs);
}
/**
 * @param {?} configs
 * @return {?}
 */
export function createConfigs(configs) {
    /** @type {?} */
    var counterConfigs;
    if (configs instanceof Array) {
        counterConfigs = configs;
    }
    else {
        counterConfigs = [(/** @type {?} */ (configs))];
    }
    return counterConfigs.map((/**
     * @param {?} config
     * @return {?}
     */
    function (config) { return Object.assign(new YandexCounterConfig(), config); }));
}
/**
 * @param {?} counterConfigs
 * @param {?} platformId
 * @return {?}
 */
export function appInitializerFactory(counterConfigs, platformId) {
    if (isPlatformBrowser(platformId)) {
        return insertMetrika.bind(null, counterConfigs);
    }
    return (/**
     * @return {?}
     */
    function () { return 'none'; });
}
/**
 * @param {?} counterConfigs
 * @return {?}
 */
export function insertMetrika(counterConfigs) {
    /** @type {?} */
    var name = 'yandex_metrika_callbacks2';
    window[name] = window[name] || [];
    window[name].push((/**
     * @return {?}
     */
    function () {
        var e_1, _a;
        try {
            try {
                for (var counterConfigs_1 = tslib_1.__values(counterConfigs), counterConfigs_1_1 = counterConfigs_1.next(); !counterConfigs_1_1.done; counterConfigs_1_1 = counterConfigs_1.next()) {
                    var config = counterConfigs_1_1.value;
                    createCounter(config);
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (counterConfigs_1_1 && !counterConfigs_1_1.done && (_a = counterConfigs_1.return)) _a.call(counterConfigs_1);
                }
                finally { if (e_1) throw e_1.error; }
            }
        }
        catch (e) { }
    }));
    /** @type {?} */
    var n = document.getElementsByTagName('script')[0];
    /** @type {?} */
    var s = document.createElement('script');
    s.type = 'text/javascript';
    s.async = true;
    /** @type {?} */
    var alternative = counterConfigs.find((/**
     * @param {?} config
     * @return {?}
     */
    function (config) { return config.alternative; }));
    if (alternative) {
        s.src = 'https://cdn.jsdelivr.net/npm/yandex-metrica-watch/tag.js';
    }
    else {
        s.src = 'https://mc.yandex.ru/metrika/tag.js';
    }
    /** @type {?} */
    var insetScriptTag = (/**
     * @return {?}
     */
    function () { return n.parentNode.insertBefore(s, n); });
    if (((/** @type {?} */ (window))).opera === '[object Opera]') {
        document.addEventListener('DOMContentLoaded', insetScriptTag, false);
    }
    else {
        insetScriptTag();
    }
    return name;
}
/**
 * @param {?} config
 * @return {?}
 */
export function createCounter(config) {
    window[getCounterNameById(config.id)] = new Ya.Metrika2(config);
}
/**
 * @param {?} id
 * @return {?}
 */
export function getCounterNameById(id) {
    return "yaCounter" + id;
}
