// ng-yandex-metrika/esm2015/lib/ng-yandex-metrika.config.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InjectionToken } from '@angular/core';
/** @type {?} */
export const DEFAULT_COUNTER_ID_AOT = new InjectionToken('DEFAULT_COUNTER_ID_AOT');
/** @type {?} */
export const DEFAULT_COUNTER_ID = new InjectionToken('DEFAULT_COUNTER_ID');
/** @type {?} */
export const YANDEX_COUNTERS_CONFIGS_AOT = new InjectionToken('YANDEX_COUNTERS_CONFIGS_AOT');
/** @type {?} */
export const YANDEX_COUNTERS_CONFIGS = new InjectionToken('YANDEX_COUNTERS_CONFIGS');
/**
 * @record
 */
export function CounterConfig() { }
if (false) {
    /** @type {?} */
    CounterConfig.prototype.id;
    /** @type {?|undefined} */
    CounterConfig.prototype.params;
    /** @type {?|undefined} */
    CounterConfig.prototype.clickmap;
    /** @type {?|undefined} */
    CounterConfig.prototype.trackLinks;
    /** @type {?|undefined} */
    CounterConfig.prototype.accurateTrackBounce;
    /** @type {?|undefined} */
    CounterConfig.prototype.webvisor;
    /** @type {?|undefined} */
    CounterConfig.prototype.trackHash;
    /** @type {?|undefined} */
    CounterConfig.prototype.ut;
    /** @type {?|undefined} */
    CounterConfig.prototype.ecommerce;
    /** @type {?|undefined} */
    CounterConfig.prototype.triggerEvent;
    /** @type {?|undefined} */
    CounterConfig.prototype.alternative;
}
export class YandexCounterConfig {
    constructor() {
        this.clickmap = true;
        this.trackLinks = true;
        this.accurateTrackBounce = true;
        this.webvisor = false;
        this.trackHash = true;
        this.ut = 'noindex';
    }
}
if (false) {
    /** @type {?} */
    YandexCounterConfig.prototype.id;
    /** @type {?} */
    YandexCounterConfig.prototype.params;
    /** @type {?} */
    YandexCounterConfig.prototype.clickmap;
    /** @type {?} */
    YandexCounterConfig.prototype.trackLinks;
    /** @type {?} */
    YandexCounterConfig.prototype.accurateTrackBounce;
    /** @type {?} */
    YandexCounterConfig.prototype.webvisor;
    /** @type {?} */
    YandexCounterConfig.prototype.trackHash;
    /** @type {?} */
    YandexCounterConfig.prototype.ut;
    /** @type {?} */
    YandexCounterConfig.prototype.ecommerce;
    /** @type {?} */
    YandexCounterConfig.prototype.triggerEvent;
    /** @type {?} */
    YandexCounterConfig.prototype.alternative;
}
